tests/analyser/analyser.cpp

```diff
@@ -389,24 +434,553 @@ TEST(Analyser, unsuitablyConstrained)
+    const std::vector<libcellml::CellmlElementType> expectedCellmlElementTypes = {
+        libcellml::CellmlElementType::VARIABLE,
+    };
+    const std::vector<libcellml::Issue::Level> expectedLevels = {
+        libcellml::Issue::Level::MESSAGE,
+    };
+    const std::vector<libcellml::Issue::ReferenceRule> expectedReferenceRules = {
+        libcellml::Issue::ReferenceRule::UNSPECIFIED,
+    };
+
+    auto analyser = libcellml::Analyser::create();
+
+    analyser->addExternalVariable(libcellml::AnalyserExternalVariable::create(model->component("membrane")->variable("V")));
+    analyser->addExternalVariable(libcellml::AnalyserExternalVariable::create(other_model->component("membrane")->variable("V")));
+
+    EXPECT_EQ(size_t(2), analyser->externalVariableCount());
+
+    analyser->analyseModel(model);
+
+    EXPECT_EQ_ISSUES_CELLMLELEMENTTYPES_LEVELS_REFERENCERULES(expectedIssues, expectedCellmlElementTypes, expectedLevels, expectedReferenceRules, analyser);
+}
+
+TEST(Analyser, removeExternalVariableByIndex)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    auto analyser = libcellml::Analyser::create();
+
+    EXPECT_FALSE(analyser->removeExternalVariable(0));
+
+    analyser->addExternalVariable(libcellml::AnalyserExternalVariable::create(model->component("membrane")->variable("V")));
+
+    EXPECT_TRUE(analyser->removeExternalVariable(0));
+    EXPECT_FALSE(analyser->removeExternalVariable(1));
+}
+
+TEST(Analyser, removeExternalVariableByName)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    auto analyser = libcellml::Analyser::create();
+
+    EXPECT_FALSE(analyser->removeExternalVariable(model, "membrane", "V"));
+
+    analyser->addExternalVariable(libcellml::AnalyserExternalVariable::create(model->component("membrane")->variable("V")));
+
+    EXPECT_TRUE(analyser->removeExternalVariable(model, "membrane", "V"));
+    EXPECT_FALSE(analyser->removeExternalVariable(model, "membrane", "X"));
+}
+
+TEST(Analyser, removeExternalVariableByPointer)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("generator/hodgkin_huxley_squid_axon_model_1952/model.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    auto analyser = libcellml::Analyser::create();
+    auto externalVariable = libcellml::AnalyserExternalVariable::create(model->component("membrane")->variable("V"));
+
+    EXPECT_FALSE(analyser->removeExternalVariable(externalVariable));
+
+    analyser->addExternalVariable(externalVariable);
+
+    EXPECT_TRUE(analyser->removeExternalVariable(externalVariable));
+    EXPECT_FALSE(analyser->removeExternalVariable(nullptr));
+}
+
```

Comment:
Do you need a test for where the external variable is deleted after it's been added to the analyser?

Reply:
Do you mean having the following:
```
EXPECT_TRUE(analyser->removeExternalVariable(externalVariable));
EXPECT_FALSE(analyser->removeExternalVariable(externalVariable));
EXPECT_FALSE(analyser->removeExternalVariable(nullptr));
```
rather than the above? If so, done in commit 1416e92ce.